backend/bin/e2e-seed-new.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+/* eslint no-console: off */
+/* eslint import/no-extraneous-dependencies: off */
+
+// This script can be run manually via yarn to prepare data for the app
+// - It publishes a "branch-created" event onto the rabblerouser kinesis stream
+// - It publishes an "admin-created" event onto the rabblerouser kinesis stream
+// - In development, it also ensures that the kinesis stream and event S3 bucket have been created
+//   (In production, these are created by terraform before the app is deployed)
+
+const AWS = require('aws-sdk');
+const branchesController = require('../src/controllers/branchesController');
+const adminController = require('../src/controllers/adminController');
+const adminType = require('../src/security/adminType');
+const uuid = require('node-uuid');
+const branchId = uuid();
+
+const createStream = () => {
+  console.log('Creating kinesis stream for development');
+
+  const kinesis = new AWS.Kinesis({
+    endpoint: process.env.KINESIS_ENDPOINT,
+    region: 'ap-southeast-2',
+    accessKeyId: 'FAKE',
+    secretAccessKey: 'ALSO FAKE',
+  });
+
+  const StreamName = process.env.STREAM_NAME;
+  return kinesis.createStream({ StreamName, ShardCount: 1 }).promise().then(
+    () => console.log(`${StreamName} created`),
+    err => {
+      // Swallow these errors, but re-throw all others
+      if (err.message.includes('already exists')) {
+        console.log(`${StreamName} already exists`);
+        return;
+      }
+      throw new Error(`Could not create stream: ${err.message}`);
+    }
+  );
+};
+
+const createArchiveBucket = () => {
+  console.log('Creating archive S3 bucket for development');
+
+  const s3 = new AWS.S3({
+    endpoint: process.env.S3_ENDPOINT,
+    region: 'ap-southeast-2',
+    accessKeyId: 'FAKE',
+    secretAccessKey: 'ALSO FAKE',
+  });
+
+  return s3.createBucket({ Bucket: process.env.ARCHIVE_BUCKET }).promise().then(
+    () => console.log(`${process.env.ARCHIVE_BUCKET} created`),
+    err => { throw new Error(`Could not create bucket: ${err.message}`); }
+  );
+};
+
+const createStreamAndBucket = () => {
+  if (process.env.NODE_ENV === 'production') {
+    console.log('Not creating kinesis stream or S3 bucket. These should already be there in production.');
+    return undefined;
+  }
+  return Promise.all([createStream(), createArchiveBucket()]);
+};
+
+const fakeRes = () => {
+  const res = {
+    lastStatus: null,
+    lastJson: null,
+  };
+  res.status = status => { res.lastStatus = status; return res; };
+  res.sendStatus = status => { res.lastStatus = status; return res; };
+  res.json = json => { res.lastJson = json; return res; };
+  return res;
+};
+
+const createDefaultBranch = () => {
+  console.log('Creating default branch.');
+  const req = { body: { id: branchId, name: 'A branch' } };
```

Comment:
Looking at the `branchesController`, I'm pretty sure that the id you pass in here is going to be ignore - it generates an ID for you, so you don't need to pass one in. That means you can delete lines 16 & 17 too.